### ISSUE TITLE
Error with some character like …

### ISSUE THREAD

**Issue by zimexco** [nishad/udemy-dl#20]
Hi, With udemy-dl 0.2.0 in this course : https://www.udemy.com/understand-nodejs/#/
Lecture 42 get this error : 
```
Traceback (most recent call last):
  File "/usr/local/bin/udemy-dl", line 5, in <module>
    main()
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 289, in main
    udemy_dl(username, password, link, lecture_start, lecture_end, output_dir, save_links)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 213, in udemy_dl
    directory = '{0:02d} {1!s}'.format(video['chapter_number'], video['chapter'])
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2026' in position 66: ordinal not in range(128)
```

Same with this course : https://www.udemy.com/learn-angularjs/
Lecture 5:
```
Traceback (most recent call last):
  File "/usr/local/bin/udemy-dl", line 5, in <module>
    main()
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 289, in main
    udemy_dl(username, password, link, lecture_start, lecture_end, output_dir, save_links)
  File "/Library/Python/2.7/site-packages/udemy_dl/udemy_dl.py", line 219, in udemy_dl
    filename = '{0:03d} {1!s}.mp4'.format(video['lecture_number'], video['lecture'])
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2026' in position 21: ordinal not in range(128)
```

It look like udemy-dl doesn't support the character '…',
Because the name of the lecture 5 is : `Model, View, Whatever…` and the lecture 42 is the first video of the chapter called : `Asynchronous Code, libuv, The Event Loop, Streams, Files, and more…`

**Comment by mukhlesur-millon** [comment]
this may be work, i haven't tried it.
 ``python -m pip install --upgrade https://github.com/mukhlesur-millon/udemy-dl/zipball/patch-5``
then
``python -m udemy_dl``

**Comment by nishad** [comment]
This was already addressed in https://github.com/nishad/udemy-dl-windows. Added here with https://github.com/nishad/udemy-dl/commit/0715b5c3b3d53d3702714d8567097eb054e3c204.

**Comment by mukhlesur-millon** [comment]
it was fixed here.
``python -m pip install --upgrade https://github.com/mukhlesur-millon/udemy-dl/zipball/patch-5``

**Comment by anieruddha** [comment]
modifying line 219 as follows 
filename = '{0:03d} {1!s}.mp4'.format(video['lecture_number'], video['lecture'].encode('ascii','ignore'))

fix this issue

**Comment by nishad** [comment]
Hopefully https://github.com/nishad/udemy-dl/commit/0715b5c3b3d53d3702714d8567097eb054e3c204, is a fix for your issue. This is already in the development repository.

Usually, the fixes will reflect in PyPI only after a versioned release, You can see the instructions to install from the development repository here. 

https://github.com/nishad/udemy-dl/wiki/Installing-development-version

**Comment by sirib** [comment]
adding these two lines to udemy-dl.py  after import statements worked for me.

reload(sys);
sys.setdefaultencoding("utf8")

**Comment by nishad** [comment]
Reloading sys, or messing with the terminals encoding is dangerous, and I wont recommend that in any case. 

The issue here is already fixed in https://github.com/nishad/udemy-dl/commit/0715b5c3b3d53d3702714d8567097eb054e3c204

**Comment by vineet-devin** [comment]
Is there any way to update the courses already downloaded ? I tried keeping the folder as it is and run the download again but then lecture numbers are changed so its creating duplicates. 
So I think it should check for lecture names to check for duplicates. And maybe lecture numbers can start from 1 in each folder for understanding of the order of lectures. 
PS: I did not know where to ask so apologies if i asked in the wrong place. I found that filename setting line here so asked here.
